File: solutions/2026-06-26/solution.java
// Validate IP Address
// Difficulty: Medium
// URL: https://leetcode.com/problems/validate-ip-address/

class Solution {
  public boolean validIPAddress(String IP) {
    String[] parts = IP.split(".");
    if (parts.length != 4) return false;
    for (String part : parts) {
      if (!isValidPart(part)) return false;
    }
    return true;
  }

  private boolean isValidPart(String part) {
    try {
      int num = Integer.parseInt(part);
      return num >= 0 && num <= 255 && (part.length() == 1 || part.charAt(0) != '0');
    } catch (NumberFormatException e) {
      return false;
    }
  }
}

File: solutions/2026-06-26/solution.py
# Validate IP Address
# Difficulty: Medium
# URL: https://leetcode.com/problems/validate-ip-address/

def validate_ip(ip: str) -> bool:
  def is_valid_part(part: str) -> bool:
    return 0 <= int(part) <= 255 and (len(part) == 1 or part[0] != '0')

  parts = ip.split('.')
  if len(parts) != 4:
    return False
  for part in parts:
    if not is_valid_part(part):
      return False
  return True